packages/aws-cdk/lib/settings.ts

```diff
@@ -18,23 +19,28 @@ export class Configuration {
   public readonly userConfig = new Settings();
   public readonly projectConfig = new Settings();
 
-  constructor(commandLineArguments?: Settings) {
+  constructor(commandLineArguments?: Settings | yargs.Arguments) {
+    if (commandLineArguments && !(commandLineArguments instanceof Settings)) {
+      commandLineArguments = Settings.fromCommandLineArguments(commandLineArguments);
+    }
     this.commandLineArguments = commandLineArguments || new Settings();
   }
 
   /**
    * Load all config
    */
-  public async load() {
+  public async load(): Promise<this> {
```

Comment:
I'm not seeing where this is used?

Reply:
Nowhere as of now, but I have other stuff in my pipes where I prefer to write:

```
const config = await new Configuration().load();
```

instead of

```
const config = new Configuration();
await config.load();
```